optaplanner-integration/optaplanner-spring-boot-autoconfigure/src/main/java/org/optaplanner/spring/boot/autoconfigure/SolverProperties.java

```diff
@@ -27,6 +27,10 @@
      * Defaults to {@link EnvironmentMode#REPRODUCIBLE}.
      */
     private EnvironmentMode environmentMode;
+    /**
+     * Enable daemon mode (disabled by default).
+     */
```

Comment:
Can we make that more descriptive of what daemon means? How about something like:

```
In daemon mode, non-early termination pauses the solver instead of stopping it, until the next problem fact change arrives. This is often useful for real-time planning.
Defaults to {@code false}.

```

Reply:
I like the idea of mentioning real-time planning. That gives a hint to someone that comes across this accidentally and has no idea what daemon mode could mean.

On the other hand I'd suggest to keep the documentation short and simple. This is not Javadoc. This is a text that will be used as a property description (without any processing, so things like `{@link}` that we use in Javadoc will appear there unprocessed). See https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-custom-starter-configuration-keys

Reply:
I would shorten it to
```
Enable daemon mode. This is often useful for real-time planning.
Disabled by default.
```
Wdyt?

Reply:
"Disabled by default", the other javadocs and docs use "Defaults to ...", let's be consistent.

I am ok with the shorter description, I agree that the hint to "real-time planning" is what matters.

If the usage of @link etc in the javadoc for such properties causes issues somewhere for spring boot, like in IntelliJ auto-completion in application.properties, please create a jira with a screenshot (and if you're into it a PR to fix that across all our properties ;-)

As for our docs: our docs are not javadoc, but we need a listing of all sprng-boot-starter properties somewhere (same goes for quarkus). So far just putting them in the docs is the best I figured out: refactors welcome. Meanwhile do add it so it doesn't go out of sync.

Reply:
Merged, I saw you went back to the longer description. Note that the shorter was was fine too (just was asking for the "Default to " consistency with others).

Thanks Jiri!

Reply:
Yes. In IDEA there are two modes of displaying the configuration key description.

One is in the hint window when you're choosing from available properties. In this case it only shows the first sentence. So it is good to have the first sentence as simple as possible.

![optaweb-vehicle-routing  ~-src-github com-kiegroup-optaweb-vehicle-routing  -  -optaweb-vehicle-routing-backend-src-main-resources-application properties  Back-end _002](https://user-images.githubusercontent.com/673386/77082163-fc5ecb00-69fb-11ea-8483-812e2433b5b9.png)

The second mode is when you display the documentation window for a property that is used in `application.properties`. In this case it displays the whole description and there's a plenty of space so it's actually no problem to display more details. Therefore I've decided to go with your proposal.

![optaweb-vehicle-routing  ~-src-github com-kiegroup-optaweb-vehicle-routing  -  -optaweb-vehicle-routing-backend-src-main-resources-application properties  Back-end _003](https://user-images.githubusercontent.com/673386/77082177-008ae880-69fc-11ea-856b-314af4e53122.png)